compiler/runtime.go

```diff
@@ -137,32 +146,37 @@ func (e *stringExpressionEvaluator) Type() semantic.MonoType {
 	return semantic.BasicString
 }
 
+// TODO
 func (e *stringExpressionEvaluator) Eval(ctx context.Context, scope Scope) (values.Value, error) {
 	var b strings.Builder
 	for _, p := range e.parts {
 		v, err := p.Eval(ctx, scope)
 		if err != nil {
 			return nil, err
 		}
+		defer v.Release()
```

Comment:
A defer inside of a for loop is usually wrong and I believe it's wrong here too. You either have to wrap this in a function or just manually call release. I think the second is probably better.